Draupnir 2.0.0

FSG-Cat opened the pull request:
This PR is a bit larger than a routine PR so its being submitted ahead of time. As in Draupnir 2.0.0 is not out yet at time of submitting this PR but is considered to be around the corner. 

Edit: Draupnir 2.0.0 is released and containers are on docker hub ready to go.

FSG-Cat committed:
Draupnir 2.0.0

The config getting changes all over the place is because of 2.0 having removed a lot of config options due to the code being removed.

FSG-Cat committed:
Update Draupnir Documentation to reflect state as of 2.0.0

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -27,9 +29,18 @@ pantalaimon:
   password: {{ matrix_bot_draupnir_pantalaimon_password | to_json }}
 {% endif %}
 
+# Experimental usage of the matrix-bot-sdk rust crypto.
+# This can not be used with Pantalaimon.
+# Make sure to setup the bot as if you are not using pantalaimon for this.
```
```suggestion
# Experimental usage of the matrix-bot-sdk Rust Crypto. This cannot be used with Pantalaimon.
# Make sure to configure the bot, so that it does not use Pantalaimon for it.
```

FSG-Cat replied:
This is copied straight from the upstream config and im not sure if it should be changed in the playbook. 

This specific change would most likely be rejected upstream due to violating style guidelines. Specifically the char limits as i do think this is too long as far as upstream is concerned.

luixxiul replied:
Thanks for checking!

> This specific change would most likely be rejected upstream

I'm a kind of feeling that whether it would be rejected by the upstream or not would be ultimately irrelevant to us, unless the change by our side would be referred by it, which I humbly doubt.

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -38,7 +49,7 @@ autojoinOnlyIfManager: true
 # If `autojoinOnlyIfManager` is false, only the members in this space can invite
 # the bot to new rooms.
-#acceptInvitesFromSpace: "!qporfwt:example.com"
+#acceptInvitesFromSpace: "!example:example.org"
```
```suggestion
#acceptInvitesFromSpace: "!qporfwt:example.com"
```

This is a common placeholder (despite being difficult to see it) among the documentation.

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -56,8 +67,8 @@ managementRoom: {{ matrix_bot_draupnir_management_room | to_json }}
 # Whether Draupnir should log a lot more messages in the room,
-# mainly involves "all-OK" messages, and debugging messages for when Draupnir checks bans in a room.
-#verboseLogging: false
+# mainly involves "all-OK" messages, and debugging messages for when draupnir checks bans in a room.
```
```suggestion
# mainly involves "all-OK" messages, and debugging messages for when Draupnir checks bans in a room.
```

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -56,8 +67,8 @@ managementRoom: {{ matrix_bot_draupnir_management_room | to_json }}
-# mainly involves "all-OK" messages, and debugging messages for when Draupnir checks bans in a room.
-#verboseLogging: false
+# mainly involves "all-OK" messages, and debugging messages for when draupnir checks bans in a room.
+verboseLogging: false
```
Is this intentional?

FSG-Cat replied:
Its a change that is irrelevant in the end as its the default behavior and yes i didnt see a reason to comment it out when its in the upstream config file like that. 

I did confirm with Gnuxie that its still a needed config flag as in i couldnt just remove it from the codebase like was done to other config flags that went away as part of this PR.

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -131,16 +127,19 @@ protectAllJoinedRooms: false
 # of the homeserver may be more impacted.
 backgroundDelayMS: 500
 
+# FIXME: This configuration option is currently broken in the playbook as admin APIs can not
```
```suggestion
# FIXME: This configuration option is currently broken in the playbook as admin APIs cannot
```

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -131,16 +127,19 @@ protectAllJoinedRooms: false
+#  # This only works if a local user with enough admin permissions is present in the room.
+#  enableMakeRoomAdminCommand: false
+#
+#  # Misc options for command handling and commands
```
```suggestion
# Misc options for command handling and commands
```

FSG-Cat replied:
This is done so that you can safely remove all the # markers when admin API still works again. The marked # in question is correct i do think with this context.

luixxiul replied:
I mean the last line is apparently not related to `admin` :thinking:

FSG-Cat replied:
I agree now with a second look. I am indeed blind. That was a bit silly on my part.

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -251,10 +287,13 @@ web:
     enabled: {{ matrix_bot_draupnir_abuse_reporting_enabled | to_json }}
 {% endif %}
 
+# FIXME: This configuration option is currently broken in the playbook as admin APIs can not
```
```suggestion
# FIXME: This configuration option is currently broken in the playbook as admin APIs cannot
```

luixxiul commented on roles/custom/matrix-bot-draupnir/templates/production.yaml.j2:
```diff
@@ -251,10 +287,13 @@ web:
 # instead of intercepting client calls to synapse's abuse endpoint, when that
 # isn't possible/practical.
-pollReports: false
+#pollReports: false
```
Is this intentional?

FSG-Cat replied:
Yes. its commented out due to the fact it doesnt work.

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
````diff
@@ -111,7 +123,7 @@ matrix_bot_draupnir_homeserver_url: "{{ 'http://matrix-pantalaimon:8009' if matr
 ```
 
-#### Configuration without E2EE support
+#### Configuration without Pantalaimon
````
I could not follow this change. Is "Configuration without Pantalaimon" either "Configuration with native E2EE support" or "Configuration without native E2EE support" ?

FSG-Cat replied:
This is one of those changes that must have slipped thru the cracks of not saving in vscode. I have this reworked on my desktop at home i do remember. But i will recreate it on my laptop as im away from home.

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -131,13 +143,13 @@ The first method intercepts the report API endpoint of the client-server API, wh
 You can configure additional options by adding the `matrix_bot_draupnir_configuration_extension_yaml` variable.
 
-For example, to change Draupnir's `pollReports` option to `true`, add the following configuration to your `vars.yml` file:
+For example, to change Draupnir's `acceptInvitesFromSpace` option to `!example:example.org`, add the following configuration to your `vars.yml` file:
```
```suggestion
For example, to change Draupnir's `acceptInvitesFromSpace` option to `!qporfwt:example.com`, add the following configuration to your `vars.yml` file:
```

FSG-Cat replied:
As room IDs are literally free to be whatever you want i dont see the purpose of this example value change. The TLD that might make sense. Readable room IDs with clear meanings are common in communities where users run modified Synapse that allows this.

luixxiul replied:
Please check https://github.com/spantaleev/matrix-docker-ansible-deploy/issues/3654 for the context.

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -148,13 +160,15 @@ matrix_bot_draupnir_configuration_extension_yaml: |
   # If you need something more special, you can take full control by
   # completely redefining `matrix_bot_draupnir_configuration_yaml`.
-  pollReports: true
+  acceptInvitesFromSpace: "!example:example.org"
```
```suggestion
  acceptInvitesFromSpace: "!qporfwt:example.com"
```

luixxiul commented on docs/configuring-playbook-appservice-draupnir-for-all.md:
```diff
@@ -24,9 +24,7 @@ The playbook does not create a management room for your Main Draupnir. You **nee
-<!-- TODO: enable Pantalaimon as configuring-playbook-bot-draupnir.md -->
-
-The management room has to be given an alias and be public when you are setting up the bot for the first time as the bot does not differentiate between invites and invites to the management room.
+The management room has to be given an alias and your bot has to be invited to the room. The bot previously was unable to join the management room if it was not public but this bug has been resolved.
```
```suggestion
The management room has to be given an alias, and your bot has to be invited to the room.
```

luixxiul commented on docs/configuring-playbook-appservice-draupnir-for-all.md:
```diff
@@ -36,7 +34,7 @@ As noted in the Draupnir install instructions the control room is sensitive. **A
 
 Next, set an alias to the management room.
 
-This alias can be anything you want. However, for increased security during the setup phase, it is recommended to make this alias be a random string. When it has been locked down after setup phase, you can give your room a secondary human readable alias.
+This alias can be anything you want.
```
Since the suggestion about the alias is no longer relevant, I guess you may simply remove the line.

luixxiul reviewed (commented):
Please note this is an initial round of personal review which mainly focuses on documentation and I'm neutral to configurations' validity due to my ignorance of the mechanism how Draupnir works.

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -79,9 +79,21 @@ matrix_bot_draupnir_management_room: "MANAGEMENT_ROOM_ID_HERE"
+
+### Native E2EE Support
+
+To use Native E2EE support you need to use access token based authentication from [below](####Configuration-without-E2EE-support-or-Native-E2EE).
+
+You also need to activate the support via setting `matrix_bot_draupnir_enable_experimental_rust_crypto` to `true`
+
+⚠️ **Warning**: The access token used for authentication must be obtained via means like curl. Using an access token from Element does not work with Rust Crypto. The access token in question can not have touched E2EE before use with Draupnir.
```
Also I assume the note about the Rust Crypto would be worth mentioning in the section `### Obtain an access token` above.

FSG-Cat commented on docs/configuring-playbook-appservice-draupnir-for-all.md:
```diff
@@ -85,8 +83,6 @@ ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,ensure-matrix-use
 
 **Notes**:
 
-- The `ensure-matrix-users-created` playbook tag makes the playbook automatically create the bot's user account.
-
```
This change reflects the fact that the command does nothing at all for AS mode, as it registers its own user via the config file. 

This could in theory be part of a separate PR but was included as it's part of the work done on 2.0.0 for mdad.

luixxiul replied:
Then I would like to suggest that the command just above also to be changed into below:

````
ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start
````

FSG-Cat replied:
That is a fair assessment to make that if you are only running it because of Draupnir users created is not needed.

FSG-Cat commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -168,8 +182,6 @@ ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,ensure-matrix-use
 
 **Notes**:
 
-- The `ensure-matrix-users-created` playbook tag makes the playbook automatically create the bot's user account.
-
```
The command does absolutely nothing for Draupnir as it's not on the list of services that get users automatically created. Or well if it's somehow part I haven't seen it in the list I looked at in group_vars

This could in theory be part of a separate PR but was included as it's part of the work done on 2.0.0 for mdad.

luixxiul replied:
Please see the comment above: https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/3941#discussion_r1917841515

FSG-Cat commented on roles/custom/matrix-appservice-draupnir-for-all/defaults/main.yml:
```diff
@@ -5,7 +5,7 @@
 
 # renovate: datasource=docker depName=gnuxie/draupnir
-matrix_appservice_draupnir_for_all_version: "1.87.0"
+matrix_appservice_draupnir_for_all_version: "v2.0.0"
```
D4A uses the same docker image as bot mode. The omission of the v is a screwup of mine in the previous version update.

FSG-Cat commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -79,9 +79,21 @@ matrix_bot_draupnir_management_room: "MANAGEMENT_ROOM_ID_HERE"
+
+⚠️ **Warning**: The access token used for authentication must be obtained via means like curl. Using an access token from Element does not work with Rust Crypto. The access token in question can not have touched E2EE before use with Draupnir.
+
+#### Configuring Draupnir to use Pantalaimon (Unsupported by upstream.)
```
The support is currently not removed on either side. It is simply just a bad idea but remains working for now but untested in upstream.

FSG-Cat committed:
Apply Review Feedback

Co-authored-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

FSG-Cat reviewed (commented):
Left a round of responses to feedback

Edit: Github ree. In my attempt to avoid some spam Github misplaced the comments in half the GUI.

FSG-Cat committed:
Change Room IDs found in code review to not conform to playbook standard.

Co-authored-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

FSG-Cat committed:
Further Integrate Code Review Feedback

FSG-Cat committed:
Apply remaining suggestions from code review.

Co-authored-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

FSG-Cat committed:
Apply Configuration Review Feedback

Co-authored-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

luixxiul commented:
@spantaleev I'm not yet perfectly convinced whether removing `ensure-matrix-users-created` playbook tag would be expected. Double checking would be appreciated.

spantaleev commented:
> @spantaleev I'm not yet perfectly convinced whether removing ensure-matrix-users-created playbook tag would be expected. Double checking would be appreciated.

I've checked! `ensure-matrix-users-created` does not do anything related to Draupnir. You can see that the `matrix_user_creator_users_auto` variables does not refer to anything Draupnir-related [here](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/6486362162e7b3b89aaa11e4b8a726a8b13a023e/group_vars/matrix_servers#L5809-L5858)

------------------------

It'd be great to automate user creation via the `matrix_user_creator_users_auto` Ansible variable (powered by the `ensure-matrix-users-created` playbook tag), but for this Draupnir will need to be able to log in using a username + password by itself, instead of using a username + access token.

A quick search through the [upstream repository's issues](https://github.com/the-draupnir-project/Draupnir/issues) does not reveal anyone asking for such a feature, etc. Perhaps @FSG-Cat has some internal knowledge about this.

---------------------------

It's not changed by this PR, but the **Create a management room** section seems to say:

> If you make the management room encrypted (E2EE), then you MUST enable and use Pantalaimon (see below).

Maybe this is not correct anymore?

------------------------------

I currently find the **End-to-End Encryption support** section confusing, because:

- the **Native E2EE Support** section below is still an `<h3>` (like **End-to-End Encryption support**), instead of being an `<h4>` (a sub-heading within **End-to-End Encryption support**, instead of a sibling)

- even with the above fixing, the sub-sections in the **End-to-End Encryption support** section become:
  - **Native E2EE Support** - short title: OK, good. The "You also need to activate it by setting `matrix_bot_draupnir_enable_experimental_rust_crypto` to `true`." sentence is instructional, but may be easy to miss, because it's not a full code block like all the others
  - **Configuring Draupnir to use Pantalaimon (unsupported by upstream)** - named differently (and much more verbosely) than **Native E2EE Support**, but.. it probably works. A better name may be: **Non-native E2EE Support powered by Pantalaimon (unsupported by upstream)**
  - **Configuration without Pantalaimon** - hmm.. This is not related to **End-to-End Encryption support**, yet it's currently a sub-heading (`<h4>`) of it. It should be adjusted, so that it's a sibling of **End-to-End Encryption support**.

------------

This sentence in the **Native E2EE Support** section may be better off in the Pantalaimon-specific section:

> **Note**: Draupnir does not support running with Pantalaimon as it would break all workflows that involve answering prompts with reactions.

Replacing **Note** with ⚠️ **Warning** may be of help as well. Or you can keep it a Note and use the 💡 emoji.

Perhaps an additional (more explicit) ⚠️ **Warning** about Pantalaimon being unsupported by upstream may be useful (the `(unsupported by upstream)` section heading suffix may be easier to miss).


------------------

> Rust Crypto is experimental but is considerably more stable than Pantalaimon support. It also improves over time with improvements to the element fork of matrix-bot-sdk.

"Rust Crypto" refers to Native E2EE support, I guess, but it may not be so clear to most people. "element" is capitalized incorrectly. Perhaps "Element's fork" could be a link to the actual fork you're referring to (https://github.com/element-hq/matrix-bot-sdk), while `matrix-bot-sdk` can link to the original (https://github.com/turt2live/matrix-bot-sdk, I suppose).

FSG-Cat committed:
Add Self Registration and Native Login to Draupnir

FSG-Cat committed:
Rework Draupnir Documentation to Remove Pantalaimon

FSG-Cat committed:
Set bot.draupnir as default username for the bot in config

FSG-Cat commented:
> > @spantaleev I'm not yet perfectly convinced whether removing ensure-matrix-users-created playbook tag would be expected. Double checking would be appreciated.
> 
> I've checked! `ensure-matrix-users-created` does not do anything related to Draupnir. You can see that the `matrix_user_creator_users_auto` variables does not refer to anything Draupnir-related [here](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/6486362162e7b3b89aaa11e4b8a726a8b13a023e/group_vars/matrix_servers#L5809-L5858)
> 
> It'd be great to automate user creation via the `matrix_user_creator_users_auto` Ansible variable (powered by the `ensure-matrix-users-created` playbook tag), but for this Draupnir will need to be able to log in using a username + password by itself, instead of using a username + access token.
> 
> A quick search through the [upstream repository's issues](https://github.com/the-draupnir-project/Draupnir/issues) does not reveal anyone asking for such a feature, etc. Perhaps @FSG-Cat has some internal knowledge about this.
> 
> It's not changed by this PR, but the **Create a management room** section seems to say:
> 
> > If you make the management room encrypted (E2EE), then you MUST enable and use Pantalaimon (see below).
> 
> Maybe this is not correct anymore?
> 
> I currently find the **End-to-End Encryption support** section confusing, because:
> 
>     * the **Native E2EE Support** section below is still an `<h3>` (like **End-to-End Encryption support**), instead of being an `<h4>` (a sub-heading within **End-to-End Encryption support**, instead of a sibling)
> 
>     * even with the above fixing, the sub-sections in the **End-to-End Encryption support** section become:
>       
>       * **Native E2EE Support** - short title: OK, good. The "You also need to activate it by setting `matrix_bot_draupnir_enable_experimental_rust_crypto` to `true`." sentence is instructional, but may be easy to miss, because it's not a full code block like all the others
>       * **Configuring Draupnir to use Pantalaimon (unsupported by upstream)** - named differently (and much more verbosely) than **Native E2EE Support**, but.. it probably works. A better name may be: **Non-native E2EE Support powered by Pantalaimon (unsupported by upstream)**
>       * **Configuration without Pantalaimon** - hmm.. This is not related to **End-to-End Encryption support**, yet it's currently a sub-heading (`<h4>`) of it. It should be adjusted, so that it's a sibling of **End-to-End Encryption support**.
> 
> 
> This sentence in the **Native E2EE Support** section may be better off in the Pantalaimon-specific section:
> 
> > **Note**: Draupnir does not support running with Pantalaimon as it would break all workflows that involve answering prompts with reactions.
> 
> Replacing **Note** with ⚠️ **Warning** may be of help as well. Or you can keep it a Note and use the 💡 emoji.
> 
> Perhaps an additional (more explicit) ⚠️ **Warning** about Pantalaimon being unsupported by upstream may be useful (the `(unsupported by upstream)` section heading suffix may be easier to miss).
> 
> > Rust Crypto is experimental but is considerably more stable than Pantalaimon support. It also improves over time with improvements to the element fork of matrix-bot-sdk.
> 
> "Rust Crypto" refers to Native E2EE support, I guess, but it may not be so clear to most people. "element" is capitalized incorrectly. Perhaps "Element's fork" could be a link to the actual fork you're referring to (https://github.com/element-hq/matrix-bot-sdk), while `matrix-bot-sdk` can link to the original (https://github.com/turt2live/matrix-bot-sdk, I suppose).

I have hopefully adressed a lot of these thoughts with the Documentation Rework Commit. But i do agree that its a large thing with a lot of opinions on if its the right order of things. 

I also have added the ability for Draupnir to register it self via the user creator role and then use said information to login directly to the homeserver. 

Now as for my confidence in this logic its well not the best. I did my best but well i dont know if i did it right.

FSG-Cat committed:
Draupnir 2.0.1

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
+Draupnir can receive reports in the management room.
 
-The first method intercepts the report API endpoint of the client-server API, which requires integration with the reverse proxy in front of the homeserver. If you are using traefik, the playbook can set this up for you:
+The bot can intercept the report API endpoint of the client-server API, which requires integration with the reverse proxy in front of the homeserver. If you are using traefik, this playbook can set this up for you:
```
```suggestion
The bot can intercept the report API endpoint of the client-server API, which requires integration with the reverse proxy in front of the homeserver. If you are using Traefik, this playbook can set this up for you:
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -2,67 +2,23 @@
 Using your own account, create a new invite only room that you will use to manage the bot. This is the room where you will see the status of the bot and where you will send commands to the bot, such as the command to ban a user from another room. Anyone in this room can control the bot so it is important that you only invite trusted users to this room.
 
-If you make the management room encrypted (E2EE), then you MUST enable and use Pantalaimon (see [below](#configuration-with-e2ee-support)).
+If you make the management room encrypted (E2EE), then you need to enable Native E2EE (see [below](###Native-E2EE-Support)).
```
```suggestion
If you make the management room encrypted (E2EE), then you need to enable the native E2EE support (see [below](#native-e2ee-support)).
```

I'd suggest to check whether the internal link is not broken when changing it :wink:

luixxiul commented:
The sections from `End-to-End Encryption support` to `Migrating from Mjolnir (Only required if migrating)` are supposed to be subsections of  `Adjusting the playbook configuration`.

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
-# Endpoint URL that Draupnir could use to fetch events related to reports (client-server API and /_synapse/),
-# only set this to the public-internet homeserver client API URL, do NOT set this to the pantalaimon URL.
-matrix_bot_draupnir_raw_homeserver_url: "{{ matrix_addons_homeserver_client_api_url }}"
+To expose the APIs publicly, add the following configuration to your `inventory/host_vars/matrix.example.com/vars.yml` file:
```
```suggestion
To expose the APIs publicly, add the following configuration to your `vars.yml` file:
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
````diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
-matrix_bot_draupnir_pantalaimon_username: "bot.draupnir"
-matrix_bot_draupnir_pantalaimon_password: "PASSWORD_FOR_THE_BOT"
+```yaml
+#Enables Native E2EE Support in Draupnir
````
```suggestion
# Enables the native E2EE Support
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
 
-# Tell Draupnir to use Pantalaimon
-matrix_bot_draupnir_pantalaimon_use: true
+#### Adjust the playbook configuration to enable Native E2EE
```
```suggestion
To enable the native E2EE support, add the following configuration to your `vars.yml` file:
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
````diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
 ```
 
-The playbook's `group_vars` will configure other required settings. If using this role separately without the playbook, you also need to configure the two URLs that Draupnir uses to reach the homeserver, one through Pantalaimon and one "raw". This example is taken from the playbook's `group_vars`:
+## Make sure the account is free from rate limiting
````
```suggestion
### Make sure the account is free from rate limiting
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
+
+⚠️ **Warning**: Access tokens are sensitive information. Do not include them in any bug reports, messages, or logs. Do not share the access token with anyone.
+
+## Abuse Reports
```
```suggestion
### Abuse Reports
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
+The other method polls an Synapse Admin API endpoint, hence it is available only if using Synapse and if the Draupnir user is an admin (see [above](#register-the-bot-account)). To enable it, set `pollReports: true` on `vars.yml` file as below. 
+-->
+
+## Extending the configuration
```
```suggestion
### Extending the configuration
```

luixxiul commented on docs/configuring-playbook-bot-draupnir.md:
```diff
@@ -72,72 +28,85 @@ To enable the bot, add the following configuration to your `vars.yml` file. Make
+## End-to-End Encryption support
 
-Decide whether you want Draupnir to be capable of operating in end-to-end encrypted (E2EE) rooms. This includes the management room and the moderated rooms.
+Decide whether you want to support having an Encrypted management room or not. Draupnir can still protect encrypted rooms without encryption support enabled.
 
-To support E2EE, Draupnir needs to [use Pantalaimon](configuring-playbook-pantalaimon.md).
+Refer to Draupnir's [Documentation](https://the-draupnir-project.github.io/draupnir-documentation/moderator/managing-protected-rooms#protecting-encrypted-rooms) for more information on why you might or might not care about encryption support for protected rooms.
 
-#### Configuration with E2EE support
+**Note**: Draupnir does not support running with Pantalaimon as it would break all workflows that involve answering prompts with reactions.
```
This might as well to be moved above `Adjusting the playbook configuration`.

FSG-Cat committed:
Integrate Review Feedback on Structure of Docs

Co-authored-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

FSG-Cat committed:
Further Restructure Docs and tweak variables in response.

spantaleev committed:
Only auto-create draupnir user if a password has been set

The Draupnir role supports configuring it with either an access token or with a password.

When a password is not assigned (which means the access token mode is used), the user is to be created manually.

spantaleev committed:
Add ensure-matrix-users-created tag

Now that the Draupnir user may be auto-created in certain configurations (if a password is assigned), it's useful to have the tag there.
